### PR TITLE
fix for normals on two-sided materials with negative scale

### PR DESCRIPTION
**Pull request by raytranuk**
Fixes https://github.com/playcanvas/engine/issues/2414
Fixes https://github.com/playcanvas/engine/issues/2374

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by Maksims** [comment]
Will it be ok for front / back faces ignore that, and only no culling to apply logic for negative scale?

**Comment by raytranuk** [comment]
> Will it be ok for front / back faces ignore that, and only no culling to apply logic for negative scale?

Yes - this fix is only necessary for materials with two sided lighting - other materials don't have the same problem even when applied on meshes with negative scale.

File: src/graphics/program-lib/programs/standard.js [hunk 1]
```diff
@@ -1328,20 +1328,22 @@ var standard = {
...
+        if (options.twoSidedLighting) code += "uniform float twoSidedLightingNegScaleFactor;\n";
+
         // FRAGMENT SHADER BODY
 
         code = this._fsAddStartCode(code, device, chunks, options);
 
         if (needsNormal) {
             if (options.twoSidedLighting) {
-                code += "   dVertexNormalW = gl_FrontFacing ? vNormalW : -vNormalW;\n";
+                code += "   dVertexNormalW = gl_FrontFacing ? vNormalW*twoSidedLightingNegScaleFactor : -vNormalW*twoSidedLightingNegScaleFactor;\n";
```
Comment (mvaligursky) [inline src/graphics/program-lib/programs/standard.js:1339]: maybe add some spaces around operators?

Comment (raytranuk) [reply src/graphics/program-lib/programs/standard.js]: spaces added - thanks!

**Comment by raytranuk** [comment]
NB. Passed (relevant) @slimbuck's engine visual tests.

**Review by mvaligursky — APPROVED** [review]
looks good to me

**Pull request merged**